tests/utils/test__select_column.py

```diff
@@ -0,0 +1,248 @@
+
+
+def test_slice_dtypes(df):
+    """
+    Raise ValueError if `columns_to_select` is a slice instance
+    and either the start value or the stop value is not a string,
+    or the step value is not an integer.
+    """
+    with pytest.raises(ValueError):
+        _select_columns(slice(1, "M_end_date_2"), df)
+    with pytest.raises(ValueError):
+        _select_columns(slice("id", 2), df)
+    with pytest.raises(ValueError):
+        _select_columns(slice("id", "M_end_date_2", "3"), df)
+
+
+def test_slice_presence(df):
+    """
+    Raise ValueError if `columns_to_select` is a slice instance
+    and either the start value or the end value is not present
+    in the dataframe.
+    """
+    with pytest.raises(ValueError):
+        _select_columns(slice("Id", "M_start_date_1"), df)
+    with pytest.raises(ValueError):
+        _select_columns(slice("id", "M_end_date"), df)
+
+
+def test_callable(df):
+    """
+    Check that error is raised if `columns_to_select` is a
+    callable, and at lease one Series has a wrong data type
+    that makes the callable unapplicable.
+    """
+    with pytest.raises(TypeError):
+        _select_columns(object, df)
+
+
+def test_callable_returns_Series(df):
+    """
+    Check that error is raised if `columns_to_select` is a
+    callable, and returns a Series.
+    """
+    with pytest.raises(ValueError):
+        _select_columns(lambda x: x + 1, df)
+
+
+def test_callable_no_match(df):
+    """
+    Raise ValueError if `columns_to_select` is a callable, and
+    no boolean results are returned, when the callable is
+    applied to each series in the dataframe.
+    """
+    with pytest.raises(ValueError):
+        _select_columns(pd.api.types.is_float_dtype, df)
+
+    with pytest.raises(ValueError):
+        _select_columns(lambda x: "Date" in x.name, df)
+
+
+def test_regex_presence(df):
+    """
+    Raise ValueError if `columns_to_select` is a regex
+    and none of the column names match.
+    """
+    with pytest.raises(ValueError):
+        _select_columns(re.compile(r"^\d+"), df)
+
+
+def test_strings(df1):
+    assert _select_columns("id", df1) == ["id"]
+    assert _select_columns("*type*", df1) == [
+        "type",
+        "type1",
+        "type2",
+        "type3",
+    ]
+
+
+def test_slice(df1):
+    assert_index_equal(
+        _select_columns(slice("code", "code2"), df1),
+        df1.loc[:, slice("code", "code2")].columns,
+    )
+    assert_index_equal(
+        _select_columns(slice("code2", None), df1),
+        df1.loc[:, slice("code2", None)].columns,
+    )
+    assert_index_equal(
+        _select_columns(slice(None, "code2"), df1),
+        df1.loc[:, slice(None, "code2")].columns,
+    )
+    assert_index_equal(_select_columns(slice(None, None), df1), df1.columns)
+    assert_index_equal(
+        _select_columns(slice(None, None, 2), df1),
+        df1.loc[:, slice(None, None, 2)].columns,
+    )
+
+
+def test_callable_data_type(df1):
```

Comment:
Missing docstrings for intent of the test function. Going to copy/paste below, pardon me please.